MQ2FeedMe.cpp

```diff
@@ -79,38 +253,50 @@ void ReadList(std::list<std::string>* MyList, PCHAR fSec)
 {
 	char Buffer[MAX_STRING*10];
 	MyList->clear();
-	if(GetPrivateProfileString(fSec,NULL,"",Buffer,MAX_STRING*10,INIFileName)) {
+
+	if(GetPrivateProfileString(fSec, NULL, "", Buffer, MAX_STRING * 10 , INIFileName))
+	{
 		char  szTemp[MAX_STRING];
-		PCHAR pBuffer=Buffer;
-		while (pBuffer[0]!=0) {
-			GetPrivateProfileString(fSec,pBuffer,"",szTemp,MAX_STRING,INIFileName);
-			if(szTemp[0]!=0) MyList->push_back(std::string(szTemp));
-			pBuffer+=strlen(pBuffer)+1;
+		char* pBuffer = Buffer;
+
+		while (pBuffer[0] !=0 )
+
+		{
+			GetPrivateProfileString(fSec, pBuffer, "", szTemp, MAX_STRING, INIFileName);
+			if (szTemp[0] != 0)
+			{
+				MyList->push_back(std::string(szTemp));
```

Comment:
When should you use push_back and when should you use emplace_back?

Reply:
this should be emplace_back.